api/src/main/java/org/eclipse/microprofile/openapi/models/media/Schema.java

```diff
@@ -1949,4 +1949,43 @@ default Schema contentSchema(Schema contentSchema) {
         return this;
     }
 
+    /**
+     * Returns whether this Schema is a boolean schema.
+     * <p>
+     * If this property is not {@code null}, then all other properties are ignored and the schema will be represented by
+     * a boolean {@code true} or {@code false} value.
+     *
+     * @return the boolean value of this schema, or {@code null} if it is not a boolean schema
+     * @since 4.0
+     */
+    Boolean getBooleanSchema();
+
+    /**
+     * Sets this schema to a boolean value.
+     * <p>
+     * If this property is not {@code null}, then all other properties are ignored and the schema will be represented by
+     * a boolean {@code true} or {@code false} value.
+     *
+     * @param booleanSchema
+     *            the boolean value of this schema, or {@code null} if it is not a boolean schema
+     * @since 4.0
+     */
+    void setBooleanSchema(Boolean booleanSchema);
```

Comment:
As written, this Javadoc doesn't really describe what other methods do when the schema is a boolean schema. It says "other properties are ignored", which could mean that setting them is ignored, or that they can be set, but they just won't be output.

Reasonable options would be:
- Any other properties set are lost when `setBooleanSchema` is called
  - Calling other setters could either do nothing or throw an exception
  - Calling other getters could either return `null` or throw an exception
- All other properties are kept until the object is serialized, so that `setBooleanSchema(null)` can be called to turn the schema back into an object

Reply:
Do you think this needs to go in the Javadoc or are we happy leaving it vague?